### ISSUE TITLE
JAX scipy

### ISSUE THREAD

**Issue by jecampagne** [f0uriest/interpax#6]
Hi, your lib is quite nice ! 

I have a question: JAX has a scipy module lacking an interpolate equivalent while your lib is close to file the gap. Do you plane to make a JAX PR? 

In jax.scipy the idea is to get the same API as scipy. I have been able to make available JAX scipy.signal.fftconvolve. Even, if I had made some tests before entering in the PR work, it was a quite "long" process to conform to JAX lib coding style (at least for me). 

Along this line, I am not sure that the use a **equinox** lib is necessary and would be questioned, as this is a third party lib which is dealing with NN. 
What do you think?

Best.

**Comment by f0uriest** [comment]
As far as I understand based on the recent [JEP](https://jax--18137.org.readthedocs.build/en/18137/jep/18137-numpy-scipy-scope.html), most of ``scipy.interpolate`` is "out of scope" for the main JAX library, as in there are no plans to implement and they suggest moving it to a third party library (like interpax).

I'm certainly open to combining things if that changes, but for now I figured make it standalone.

I'm also definitely open to making the API more 1-1 with scipy, what I have now was created for other projects that didn't need to follow the scipy API.

wrt equinox, while it was initially primarily a NN library it has since become the core of a number of scientific computing libraries like [diffrax](https://github.com/patrick-kidger/diffrax), [lineax](https://github.com/google/lineax), and [optimistix](https://github.com/patrick-kidger/optimistix). The main thing it provides are callable pytrees, which are kind of perfect for the scipy-like API of having interpolator classes/objects. Using equinox allows those objects to work with all the regular JAX transformations like jit,grad,vmap etc.

**Comment by jecampagne** [comment]
Thanks for the JEP link, I was not aware of and it was a subject of discussion with some colleagues. Concerning equinox, I was aware that it becomes a sort of "standard" even if it would be nice to get a lib not dealing with the NN part (but this is an other story).
Concerning, jit/vmap... 

It would be a good point if we can test interpax wrt "jacrev" as this is a trick gradient operation that is a critical point for software dealing with backprop (eg. bayesian model optimisation).

In any case if you need help... point to me a specific pb to test. Bye.
